model/task/db.go

```diff
@@ -173,6 +173,24 @@ func ByIdsBuildAndStatus(taskIds []string, buildId string, statuses []string) db
 	})
 }
 
+// ByStaleRunningTask creates a query that finds any running tasks
+// whose last heartbeat was at least the specified threshold ago, or
+// that has been dispatched but hasn't started in twice that long.
+func ByStaleRunningTask(staleness time.Duration) db.Q {
```

Comment:
It's strange to me that both this function and `FindStaleRunningTasks` exist. What's the reason for this?

Reply:
The aggregation uses the host collection and does a "left join" whereas this just queries the task collection, so they will return different results. There might be a hacky way to make the aggregation cover both with multiple $lookups but I felt this was cleaner